packages/hdwallet-gridplus/src/gridplus.ts

```diff
@@ -367,8 +377,45 @@ export class GridPlusHDWallet
 
     const { external, internal } = await this.client.fetchActiveWallet();
 
-    if (!external.uid.equals(ZERO_BUFFER)) return external.uid.toString("hex");
-    if (!internal.uid.equals(ZERO_BUFFER)) return internal.uid.toString("hex");
+    if (isSafecardConnected(external.uid)) return external.uid.toString("hex");
+    if (isSafecardConnected(internal.uid)) return internal.uid.toString("hex");
+  }
+
+  /**
+   * Validate that the currently active wallet matches expectations
+   * Throws if expectedActiveWalletId is provided and doesn't match
+   */
+  public async validateActiveWallet(
+    expectedActiveWalletId?: string,
+    expectedType?: SafeCardType
+  ): Promise<{
+    activeWalletId: string;
+    type: SafeCardType;
+  }> {
+    if (!this.client) throw new Error("Device not connected");
+
+    const activeWallets = await this.client.fetchActiveWallet();
+
+    // Determine active wallet type (external SafeCard takes priority)
```

Comment:
i.e
- if both are inserted, communicating w gridplus endpoints will communicate with the safecard
- in other words, internal safecard wallet only works if safecard is removed, even though the device/endpoints *still* know about the existence of the internal wallet and its ID